index.js

```diff
@@ -25,6 +25,9 @@ function mewt(target) {
       return mewt(newObj)
     },
     $unset: prop => {
+      if (isA && Number.isInteger(prop)) {
+        return mewt([...target.slice(0, prop), ...target.slice(prop + 1)])
```

Comment:
Nice rundown! I think let's go for clone and splice? I will create an issue re: benchmarks.